Providers/Scripts/2.4x-2.5x/Scripts/nxOMSAgentNPMConfig.py

```diff
@@ -11,15 +11,71 @@
 import hashlib
 import codecs
 import base64
+import platform
+import shutil
 protocol = imp.load_source('protocol', '../protocol.py')
 nxDSCLog = imp.load_source('nxDSCLog', '../nxDSCLog.py')
 LG = nxDSCLog.DSCLog
 
+class IOMSAgent:
+    def start_oms_agent(self):
+        pass
+    def stop_oms_agent(self):
+        pass
+
+class OMSAgentUtil(IOMSAgent):
+    def stop_oms_agent(self):
+        if os.system('sudo /opt/microsoft/omsagent/bin/service_control stop') == 0:
+            return True
+        else:
+            LG().Log('ERROR', 'Error stopping omsagent.')
+            return False
+    def start_oms_agent(self):
+        if os.system('sudo /opt/microsoft/omsagent/bin/service_control start') == 0:
+            return True
+        else:
+            LG().Log('ERROR', 'Error starting omsagent.')
+            return False
+
+class INPMAgent:
+    def binary_setcap(self):
+        pass
+
+class NPMAgentUtil(IOMSAgent):
+    def binary_setcap(self, binaryPath):
+        if os.system('sudo /opt/microsoft/omsconfig/Scripts/NPMAgentBinaryCap.sh ' + binaryPath) == 0:
```

Comment:
in the MOF template in AMS, you also need to check the agent version before deliver the nxOMSAgentNPMResource.